components/03-modules/popup/popup.js

```diff
@@ -41,69 +48,53 @@
 
     show() {
       this.popup.classList.add(this.active);
+      this.popup.open = true;
+      this.popupContent.focus();
       this.position();
+      window.addEventListener('keydown', this.keyboardClosePopup.bind(this));
     }
 
     hide() {
       this.popup.classList.remove(this.active);
+      this.button.focus();
+      this.popup.open = false;
     }
 
     position() {
-      const arrow        = this.arrowSelector.getBoundingClientRect(),
-            buttonCoords = this.button.getBoundingClientRect();
-
-      this.popup.style.setProperty('top', `${buttonCoords.height + arrow.height / 2}px`, '');
+      const buttonCoords = this.button.getBoundingClientRect();
+      this.popup.style.setProperty('top', `${buttonCoords.height}px`, '');
       this.popup.style.setProperty('left', `${buttonCoords.left}px`, '');
-      this.isOutOfscreen(buttonCoords);
+      this.isOutOfscreen();
     }
 
-    isOutOfscreen(buttonCoords) {
+    isOutOfscreen() {
       const popupContentCoords = this.popup.getBoundingClientRect();
 
       if (window.matchMedia('(max-width: 480px)').matches) {
         const calcLeft = this.button.getBoundingClientRect().left * -1;
 
         this.popup.classList.add('popup--full-width');
         this.popup.style.setProperty('left', `${calcLeft}px`, '');
-
-        this.arrowSetPosition(buttonCoords.width, true);
       }
       else if (popupContentCoords.right >= this.body.clientWidth) {
         this.popup.classList.remove('popup--full-width');
```

Comment:
we have `this.active` and there is modifier. Can we add this one too?

Reply:
corrected